### ISSUE TITLE
parallel=True and if-condition

### ISSUE THREAD

**Issue by piotrbartman** [numba/numba#5279]
## Reporting a bug

Numba version: 0.48.0

```Python
import numba
import numpy as np


@numba.njit(parallel=True)
def bug(swap, arr):
    j, k = 0, 1
    if swap:
        j, k = k, j
    arr[:, k] = 3
    return arr


if __name__ == '__main__':
    print("numba:  ",         bug(True, np.array([[1, 2]])))
    print("py_func:", bug.py_func(True, np.array([[1, 2]])))
```

Output:
```
numba:   [[1 3]]
py_func: [[3 2]]
```

If we replace `arr[:, k] = 3` with `arr[0, k] = 3` it works properly (with warning "no transformation for parallel execution was possible.")

**Comment by esc** [comment]
@piotrbartman thanks for reporting this to the Numba issue tracker. I can confirm that this is probably a bug as I am able to reproduce this locally. Also note that setting `paralle=False`  also makes the code produce the correct  output.

**Comment by stuartarchibald** [comment]
cc @DrTodd13 but before looking into this, let's see if SSA just fixes it!

**Comment by piotrbartman** [comment]
In numba 0.49RC1 it fixed.

**Comment by esc** [comment]
@piotrbartman wonderful, thanks for reporting back. Does this suffice to be able to close the issue?

**Comment by piotrbartman** [comment]
Yes, thank you!

**Comment by esc** [comment]
@piotrbartman probably not, since those tests would have failed due to lack of SSA. Unless of course those tests were skipped.

**Comment by stuartarchibald** [comment]
Fixed in version 0.50.0+.
